server/s3_list_object_versions_action.cc

```diff
@@ -115,7 +115,29 @@ void S3ListObjectVersionsAction::validate_request() {
   request_key_marker = request->get_query_string_value("key-marker");
   s3_log(S3_LOG_DEBUG, request_id, "request_key_marker = %s\n",
          request_key_marker.c_str());
-  last_key = request_key_marker;  // As requested by user
+  if (request->has_query_param_key("version-id-marker")) {
+    request_version_id_marker =
+        request->get_query_string_value("version-id-marker");
+    s3_log(S3_LOG_INFO, request_id, "request_version_id_marker = %s\n",
+           request_version_id_marker.c_str());
+    if (request_version_id_marker.empty()) {
+      s3_log(S3_LOG_DEBUG, request_id, "version-id-marker is empty\n");
+      // TODO: Add invalid argument details to the response.
+      set_s3_error("InvalidArgument");
+      send_response_to_s3_client();
+      return;
+    } else if (request_key_marker.empty()) {
+      s3_log(S3_LOG_DEBUG, request_id, "key-marker is empty\n");
+      // TODO: Add invalid argument details to the response.
+      set_s3_error("InvalidArgument");
```

Comment:
```suggestion
      set_s3_error("InvalidArgument");
      set_s3_error_message("A version-id marker cannot be specified without a key marker.");
      set_invalid_argument("version-id-marker", request_version_id_marker);
```

Here's the result from AWS:

```txt
$ aws --debug s3api list-object-versions --bucket mybucket --version-id-marker 8e5_5Mm3btVpNYmt1rQZnFfkGM0aRgd9

An error occurred (InvalidArgument) when calling the ListObjectVersions operation: A version-id marker cannot be specified without a key marker.
```

Debug output:
```xml
<?xml version="1.0" encoding="UTF-8"?>\n
<Error>
    <Code>InvalidArgument</Code>
    <Message>A version-id marker cannot be specified without a key marker.</Message>
    <ArgumentName>version-id-marker</ArgumentName>
    <ArgumentValue>8e5_5Mm3btVpNYmt1rQZnFfkGM0aRgd9</ArgumentValue>
    <RequestId>XJZATXKZAK5DZJW6</RequestId>
    <HostId>9bn+DPs/Ec90ovM8aBQlbn3oktn/YjKXxmUGKZkuVhKzVhR4uJJQRmoWxFU1SoUdB8Hll8a20oQ=</HostId>
</Error>
```

Reply:
Done